beesdoo_shift/models/cooperative_status.py

```diff
@@ -222,9 +222,19 @@ def _state_change(self, new_state):
                 data['alert_start_time'] = False
             self.write(data)
         if new_state == 'unsubscribed' or new_state == 'resigning':
-            self.cooperator_id.sudo().write({'subscribed_shift_ids' : [(5,0,0)]})
-            #TODO: Add one day othertwise unsubscribed from the shift you were absent
-            self.env['beesdoo.shift.shift'].sudo().unsubscribe_from_today([self.cooperator_id.id], today=self.today)
+            self.cooperator_id.sudo().write({'subscribed_shift_ids': [(5, 0, 0)]})
+            # Unsubscribe from shift template if super coop
+            if self.super:
+                super_cooperator_id = self.env['res.users'].search(
+                    [('partner_id', '=', self.cooperator_id.id)])
+                super_cooperator_ids = (self.env['beesdoo.shift.template']
+                                        .search([('super_coop_id', '=',
+                                                  super_cooperator_id.id)]))
+                super_cooperator_ids.write({'super_coop_id': False})
```

Comment:
à mon avis on peut simplifier le code en récupérant directement le shift template sur le cooperator_id via le champ subscribed_shift_ids.

the write `write({'super_coop_id': False})` on the many2many field should do the trick
https://odoo-new-api-guide-line.readthedocs.io/en/latest/environment.html#many2many-one2many-behavior

Reply:
@houssine78  comme ceci ? self.cooperator_id.subscribed_shift_ids.write({'super_coop_id': False})
Ça n'a pas l'air de fonctionner

Reply:
et en itérant dessus ?

Reply:
pas mieux, on peut voir cela demain

Reply:
Pour moi le reste est bon, car dans `self.subscribed_shift_ids`, il n'y a que les shifts pour lesquels le coopérateur est inscrit comme travailleur.

Ce que fait le code pour l'instant c'est d'aller retrouver tous les créneaux (shift.template) et de retirer le coopérateur du champ super-coopérateur.

Il faudrait, je pense aussi aller rechercher tous les shifts (shift.shift) pour lesquels le coopérateur est supercoopérateur et le retirer du champ surper-coopérateur. Je pense qu'on peut accéder à ces shifts depuis l'objet shitf.template. À vérifier.

Reply:
@remytms 
Ce n'est pas déjà le cas ici ?

# If worker is Super cooperator, remove it from planning
        super_coop_ids = self.env['res.users'].search(
            [('partner_id', 'in', worker_ids), ('super', '=', True)]).ids

        if super_coop_ids:
            to_unsubscribe_super_coop = self.search(
                [('super_coop_id', 'in', super_coop_ids)] + date_domain)
            to_unsubscribe_super_coop.write({'super_coop_id': False,
                                             'is_regular': False})